### PR TITLE
Fix Social Share For mobile view.

### PR DESCRIPTION
**Pull request by amit-webkul**
## Issue Reference
#9772

### CODE REVIEW

**Comment by ashishkumar-webkul** [comment]
Checked and Found that warning pop-up is appearing when I am checking the PR, It is not compatible with browser. Please check the video for reference.

**Video**

https://github.com/bagisto/bagisto/assets/92721837/4c7e9aec-3aff-47b0-a497-cb4eff484f7c

**Comment by amit-webkul** [comment]
@ashishkumar-webkul ｓiｒ

I have checked in mobile phone the social share is working fine please find the below attachment. 

https://github.com/bagisto/bagisto/assets/121479029/498e8e19-dcd6-4706-bd1c-8b7eaf81e09c

File: packages/Webkul/SocialShare/src/Resources/views/share.blade.php [hunk 1]
```diff
@@ -28,4 +28,24 @@
 
         {!! view_render_event('bagisto.shop.products.view.share.after', ['product' => $product]) !!}
     </div>
+
+    @push('scripts')
+        <script>
+            function shareProduct() {
+                var productName = "{{ $product->name }}";
+                var productUrl = "{{ route('shop.product_or_category.index', [$product->url_key]) }}";
```
Comment (devansh-webkul) [inline packages/Webkul/SocialShare/src/Resources/views/share.blade.php:36]: Use let.

File: packages/Webkul/SocialShare/src/Resources/views/share.blade.php [hunk 2]
```diff
@@ -9,7 +9,7 @@
         <div class="hidden gap-2.5 justify-center items-center max-md:flex cursor-pointer">
             <span class="icon-share text-2xl"></span>
 
-            <a href="intent://share/#Intent;action=android.intent.action.SEND;type=text/plain;S.android.intent.extra.TEXT={{ rawurlencode($product->name . ' ' . route('shop.product_or_category.index', [$product->url_key])) }};end">
+            <a href="#" onclick="shareProduct()">
```
Comment (devansh-webkul) [inline packages/Webkul/SocialShare/src/Resources/views/share.blade.php:12]: javascript:void(0);

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/SocialShare/src/Resources/views/share.blade.php (hunk 1 above)
Comment (amit-webkul) [reply packages/Webkul/SocialShare/src/Resources/views/share.blade.php]: Done sir

File: packages/Webkul/SocialShare/src/Resources/views/share.blade.php (hunk 2 above)
Comment (amit-webkul) [reply packages/Webkul/SocialShare/src/Resources/views/share.blade.php]: Done sir

**Comment by ashishkumar-webkul** [comment]
Checked and found that the social share is working, please check the video for reference.

**Video**

https://github.com/bagisto/bagisto/assets/92721837/5bada79a-07dc-4cf2-a6e3-9e1a124356b1

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**